x-pack/plugins/infra/public/hooks/use_trial_state.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useEffect } from 'react';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+import { API_BASE_PATH as LICENSE_MANAGEMENT_API_BASE_PATH } from '../../../license_management/common/constants';
+
+export enum TrialStatusLoadState {
+  Loading = 'loading',
+  Ok = 'ok',
+  Error = 'error',
+}
+
+interface UseTrialStatusState {
+  loadState: TrialStatusLoadState;
+  isTrialAvailable: boolean;
+}
+
+export function useTrialStatus(): UseTrialStatusState {
+  const { services } = useKibana();
+
+  const [loadState, setLoadState] = useState<TrialStatusLoadState>(TrialStatusLoadState.Loading);
+  const [isTrialAvailable, setIsTrialAvailable] = useState<boolean>(false);
+
+  useEffect(() => {
+    async function fetchTrial() {
+      try {
+        const response = await services.http.get(`${LICENSE_MANAGEMENT_API_BASE_PATH}/start_trial`);
+        setIsTrialAvailable(response);
+        setLoadState(TrialStatusLoadState.Ok);
+      } catch {
+        setLoadState(TrialStatusLoadState.Error);
```

Comment:
AFAICT we don't handle the Error state? Can we render an error message if something goes wrong fetching this data?

Reply:
I'm thinking out loud.

I wonder what's best here. I get why we want to show feedback to the user about the error, but it's not really actionable. What I do right now is assume that the trial cannot be started (for whatever reason) and just point the user to the subscriptions page. I think from a user perspective that's more useful than an error.

However, we can still show a toast or something saying that "we couldn't fetch the status of your trial license".

Makes sense?